pandas/_libs/properties.pyx

```diff
@@ -42,19 +42,24 @@ cache_readonly = CachedProperty
 
 
 cdef class AxisProperty(object):
-    cdef:
+
+    cdef readonly:
         Py_ssize_t axis
 
-    def __init__(self, axis=0):
+    cdef readonly:
+        object __doc__
```

Comment:
Can this go under the same `def readonly` block as axis?

And do we want `object` or `str` for the type?

Reply:
whoops, the above `cdef readonly` is a typo, I meant to keep it as `readonly`. 
I am not fully sure if it is actually needed for `axis`, or if we can also make it `readonly` (I would think it is never modified internally, bot not sure)

For the `object`, it is what is done for `CachedProperties` above as well. My cython is not good enough to really know the consequences. I think we usually use `object` for strings in our cython code.

Reply:
If the test pass with using `readonly` for `axis`, I will combine them in one `cdef readonly`

Reply:
`object` vs `str` is pretty harmless in this case.  I'd advocate sticking with `object` for now b/c then we don't need to worry about potential py2/py3 bytes/unicode corner cases.